### PR TITLE
Add butterflies mod

### LINKED ISSUE #2090: Add butterflies mod
I would certainly like flappy butterflies as a daytime bug, hidden at night.

Since the fireflies mod has a bug net and the 'catchable' group it seems possibly best to include butterflies in that mod? Then rename the mod to 'bugs' or whatever is suitable.
We are in dev stage so don't worry about causing some unknown nodes in some worlds.

### PR DESCRIPTION
**Pull request by Ezhh**
Adds butterflies (plantlike drawtype, same approach as fireflies). They have a chance to appear close to flowers on mapgen.

@paramat I'm not sure if this needs a separate register_decoration for v6 mapgen, given that the flowers limit the placement, so I've not added one for now, but please let me know your thoughts.

Closes #2090.

### CODE REVIEW

**Comment by Fixer-007** [comment]
Can I catch them in bottles?

**Comment by paramat** [comment]
They're in the 'catchable' group so you can catch them in the bugnet that fireflies provides to add them to your inventory, then place them anywhere you want. But no bottles.

I actually don't feel fireflies and butterflies should be in mgv6, as they are decorations and new decorations are not added to mgv6 because mgv6 is stable, we would be altering the biomes.
I also don't want to encourage use of mgv6 by adding new features to it.
Anyway, since you have specifies biomes for the decorations butterflies will not appear in mgv6, which i prefer.

Will review and test.

File: mods/butterflies/README.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (paramat) [inline mods/butterflies/README.txt]: Inconsistent empty line

**Comment by Ezhh** [comment]
> Anyway, since you have specifies biomes for the decorations butterflies will not appear in mgv6...

Based on my tests, they do. The biome lines are just ignored. They don't spread everywhere like the fireflies originally did, simply because the butterflies can only appear next to flowers.

File: mods/butterflies/README.txt (hunk 1 above)
Comment (Ezhh) [reply mods/butterflies/README.txt]: Heh, sorry. I often leave the first line in my own mods. Will adjust.

**Comment by paramat** [comment]
Interesting, ok that's fine then.

**Comment by Ezhh** [comment]
Line removed.

Please feel free to test v6 before approving. It will be easy enough to prevent butterflies there or adjust them if needed.

**Comment by paramat** [comment]
They do appear in mgv6. I'd rather they and fireflies didn't though, what do you think about that?

screenshot_20180625_191141

Here's a suggestion. At head level it is almost impossible to see the flapping of the wings and the angle between them, but if placed just above ground level the flapping wings and the angle between them is visible, i would prefer them lower to look cuter, what do you think?

One issue, when i flood a butterfly the item isn't dropped for some reason, can't see why.

**Comment by Ezhh** [comment]
I did try them lower, but I really didn't like it. I wish there was a way to rotate plantlike nodes to angle things like this.

Do we want butterflies to drop on flooding? I don't think we allowed this with fireflies, and I see that kind of like killing them, so they'd be gone instead of a drop to pick up. (This also lets us avoid encouraging people to "kill" them in water to skip making a net.)

For v6, we could remove both, though in the case of fireflies it's basically changing the decision that was already made. I'll leave you to pick. I don't really see any problems adding new decorations to v6 either way, since otherwise you have to specifically exclude v6 from every new register_decoration forever and that's just a bad situation to be in.

**Comment by paramat** [comment]
Ok, maybe butterflies 1 node higher then?

Fireflies are dropped as an item on_flood:
https://github.com/minetest/minetest_game/blob/5692c15b4d19eab371a0daba2efa9f04c00b81b9/mods/fireflies/init.lua#L27
The same code is here but they just completely disappear for some reason.
I'm fine with having butterflies just 'disappear' on_flood (no droped item) if you prefer, same with fireflies. After all they disappear at night anyway.
> since otherwise you have to specifically exclude v6 from every new register_decoration forever and that's just a bad situation to be in.

Good point, here it will take extra code to exclude butterflies fom mgv6. So how about this: don't add new decorations to mgv6 if it requires extra code, but if it happens automatically like here allow it?

**Comment by paramat** [comment]
Another possibility is to keep them in the same node but move the texture as low as possible in that node, to be below eyelevel.

**Comment by Ezhh** [comment]
> Ok, maybe butterflies 1 node higher then?

I was thinker higher sounds good. Not sure which you have tested, but will play around with it a bit.

I'd like to prevent fireflies dropping on flood if you don't object... I thought that had been prevented before. I've also probably found a... bug with those bugs, but will make a separate PR for that in a bit.

**Comment by paramat** [comment]
butterflies_butterfly_white_animated

White butterfly texture for testing.

**Comment by paramat** [comment]
> I'd like to prevent fireflies dropping on flood

Fine for me, butterflies too. It's a little unpleasant to drown them, especially butterflies.

**Comment by paramat** [comment]
I prefer higher butterflies on testing.

**Comment by paramat** [comment]
:+1:

**Pull request merged**